library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64SerializerThreadpoolExecutor.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+internal object Base64SerializerThreadpoolExecutor {
```

Comment:
you could make it:
`internal object SingleInstanceThreadPoolExecutor:ExecutorsService by threadPoolExecutor` and then force it by type in the `Base64Serializer(private val executorService:SingleInstanceThreadPoolExecutor)`